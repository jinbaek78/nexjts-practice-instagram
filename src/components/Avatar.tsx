import Image from 'next/image';

type Props = {
  src: string;
  email: string;
  name?: string;
};
export default function Avatar({ src, email, name }: Props) {
  return (
    <div className="flex items-center">
      <Image
        className="rounded-full"
        src={src || ''}
        alt="avatar"
        width={80}
        height={100}
      />
      <div className="ml-5">
        <h3 className="text-xl font-bold">{email && email.split('@')[0]}</h3>
        {name && <h3 className="text-2xl text-zinc-600">{name}</h3>}
      </div>
    </div>
  );
}
