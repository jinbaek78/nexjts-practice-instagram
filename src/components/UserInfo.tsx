import { useSession } from 'next-auth/react';
import Image from 'next/image';
import Avatar from './Avatar';

export default function UserInfo() {
  const { data: session } = useSession();
  const { name, email, image } = session!.user!;

  return (
    <section className="flex flex-col gap-10">
      <Avatar src={image || ''} email={email || ''} name={name || ''} />
      <p className="text-zinc-600 text-lg">
        About﹒ Help﹒Press﹒API﹒Jobs﹒
        <br />
        Privacy﹒Terms﹒Location﹒
        <br />
        Language
      </p>
      <p className="text-zinc-700 font-semibold">
        Copyright INSTANTGARM
        <br />
        from MENTAL
      </p>
    </section>
  );
}
