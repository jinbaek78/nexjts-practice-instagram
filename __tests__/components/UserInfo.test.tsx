import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import UserInfo from '@/components/UserInfo';
import { useSession } from 'next-auth/react';
import { fakeSession } from '@/tests/mock/mock';
import Avatar from '@/components/Avatar';

jest.mock('@/components/Avatar');
jest.mock('next-auth/react', () => ({
  useSession: jest.fn(),
}));

describe('UserInfo', () => {
  const { name, email, image } = fakeSession!.user!;

  afterEach(() => {
    (useSession as jest.Mock).mockReset();
    (Avatar as jest.Mock).mockReset();
  });

  it('renders', () => {
    (useSession as jest.Mock).mockImplementation(() => ({ data: fakeSession }));
    const { container } = render(<UserInfo />);
    expect(container).toMatchSnapshot();
  });
  it('renders with user information', () => {
    (useSession as jest.Mock).mockImplementation(() => ({ data: fakeSession }));
    render(<UserInfo />);

    expect(Avatar).toHaveBeenCalledTimes(1);
    expect((Avatar as jest.Mock).mock.calls[0][0].src).toBe(image);
    expect((Avatar as jest.Mock).mock.calls[0][0].email).toBe(email);
    expect((Avatar as jest.Mock).mock.calls[0][0].name).toBe(name);
  });
});
