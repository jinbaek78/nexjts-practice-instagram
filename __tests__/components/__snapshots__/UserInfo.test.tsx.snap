// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo renders 1`] = `
<div>
  <section
    class="flex flex-col gap-10"
  >
    <p
      class="text-zinc-600 text-lg"
    >
      About﹒ Help﹒Press﹒API﹒Jobs﹒
      <br />
      Privacy﹒Terms﹒Location﹒
      <br />
      Language
    </p>
    <p
      class="text-zinc-700 font-semibold"
    >
      Copyright INSTANTGARM
      <br />
      from MENTAL
    </p>
  </section>
</div>
`;
